src/func.cc

```diff
@@ -924,6 +924,22 @@ ValuePtr builtin_cross(const Context *, const EvalContext *evalctx)
 	return ValuePtr(result);
 }
 
+ValuePtr builtin_is_undef(const Context *, const EvalContext *evalctx)
+{
+	if (evalctx->numArgs() == 1) {
+		const auto &arg =evalctx->getArgs()[0];
+		ValuePtr v;
+		if(const Lookup* lookup = dynamic_cast<const Lookup*> (arg.expr.get())){
```

Comment:
Tip: You can use `dynamic_pointer_cast` instead of unpacking the shared_ptr

Reply:
That is much nicer. And there is no need to repeat the types, so auto simplifies it further.